django/template/base.py

```diff
@@ -913,15 +913,18 @@ def _resolve_lookup(self, context):
                         try:  # method call (assuming no args required)
                             current = current()
                         except TypeError:
-                            signature = inspect.signature(current)
                             try:
-                                signature.bind()
-                            except TypeError:  # arguments *were* required
-                                current = (
-                                    context.template.engine.string_if_invalid
-                                )  # invalid method call
-                            else:
-                                raise
+                                signature = inspect.signature(current)
+                                try:
+                                    signature.bind()
+                                except TypeError:  # arguments *were* required
+                                    current = (
+                                        context.template.engine.string_if_invalid
+                                    )  # invalid method call
+                                else:
+                                    raise
+                            except ValueError:  # no signature found
+                                current = context.template.engine.string_if_invalid
```

Comment:
Is it worth having nested `try ... except`? :thinking: It should be fine to catch both:
```python
try:
    signature = inspect.signature(current)
    signature.bind()
except (
    # arguments *were* required
    TypeError,
    # no signature found
    ValueError,
):  
    current = (
        context.template.engine.string_if_invalid
    )  # invalid method call
else:
    raise
```
or use separate clauses:
```python
try:
    signature = inspect.signature(current)
except ValueError:  # no signature found
    current = context.template.engine.string_if_invalid
else:
    try:
        signature.bind()
    except TypeError:  # arguments *were* required
        current = (
            context.template.engine.string_if_invalid
        )  # invalid method call
    else:
        raise
```